### ISSUE TITLE
Better tabs implementation for entity page

### ISSUE THREAD

**Issue by dtuite** [spotify/backstage#1889]
We are working on a [GitHub Pull Requests plugin](https://github.com/RoadieHQ/backstage/pull/7) which will have a list of pull requests for each entity, and a "top level stats" widget for each entity ([as suggested here](https://github.com/spotify/backstage/issues/1808)). 

The widget can on the [entity overview page](https://backstage-demo.roadie.io/catalog/Component/sample-service/), but the list of pull requests should really go into a tab on the entity page.

We would like to contribute the tabs implementation for the entity page and thought it would be wise to have a discussion here first, in order to ensure everyone is happy.

There are at least 2 requirements I can think of:

 1. Tabs should each have a unique URL path so I can link people to a specific tab. Eg. `/catalog/Component/sample-service/ci-cd` or `/catalog/Component/sample-service/monitoring`.
 2. Tabs should only appear if there is a plugin, and metadata available, to have content in the tab.

### Related issues

 1. Sub-routing: https://github.com/spotify/backstage/issues/1042
 2. Original design: https://github.com/spotify/backstage/issues/922
 3. Plugin widgets and composability: https://github.com/spotify/backstage/issues/1536

**Comment by dtuite** [comment]
/cc @tomek-ifm 

@stefanalund assign this to me if you'd like.

**Comment by hooloovooo** [comment]
Just a heads up. There is a implementation of working tabs on the entity page in this PR. https://github.com/spotify/backstage/pull/1835
It's not ideal, but might be of interest if you're planning on doing a proper implementation.

**Comment by dtuite** [comment]
Thanks @hooloovooo . I missed that PR.

What is not ideal about the implementation in that PR? How can we help out?

**Comment by hooloovooo** [comment]
@dtuite The main issue is that there's no solution there for checking if a plugin is installed or not. I think this is meant to be handled by the upcoming widgets, but for now it doesn't handle for example a backstage setup where TechDocs is missing very well. It could be possible to make a more robust solution until widgets are in place.

**Comment by Rugvip** [comment]
As everything else, blocked by https://github.com/spotify/backstage/issues/1536 :p

**Comment by dtuite** [comment]
Got it. The work in https://github.com/spotify/backstage/pull/1835 is enough for our needs right now. It will allow us to create some tabs and put content on them. 

@stefanalund feel free to close this if large tabs improvements are tracked elsewhere.

@hooloovooo is mob/techdocs-improvements planned to be merged into master any time soon? It would be great if we could make use of those tabs.

**Comment by hooloovooo** [comment]
@dtuite I hope it'll be merged soon. We have a few issues left to resolve, but i would expect it to be merged in the next few days.

**Comment by freben** [comment]
@dtuite Would you like to close this and make new issues for additions needed on top of #1835 after it is merged, or would you prefer leaving this open for a bit longer to track it here? I'm fine with either. I don't think we have another issue yet specifically for the tabs; rather, several concepts are bunched together under #1536, and specifics like the ones you mention can be useful to track on their own to make sure we get them sorted out.

**Comment by dtuite** [comment]
@freben I have no desire to leave this open. Close it if you wish. The only reason I'm not closing it myself is because I noticed @stefanalund added it to a milestone so there may be some reason he wants it around.

**Comment by stefanalund** [comment]
Removed from the Milestone now.
